### PR TITLE
feat(react): add DataGrid component 

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Purpose

Add DataGrid Component

### Related Issues
- https://github.com/wso2/oxygen-ui/issues/2

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/components/DataGrid/DataGrid.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import {ArgsTable, Source, Story, Canvas, Meta} from '@storybook/addon-docs';
+import dedent from 'ts-dedent';
+import StoryConfig from '../../../.storybook/story-config.ts';
+import {withDesign} from '../../../.storybook/utils.ts';
```
Comment (savindi7) [inline packages/react/src/components/DataGrid/DataGrid.stories.mdx:4]: This import is not needed as we haven't included the design for this component

File: packages/react/src/components/DataGrid/DataGrid.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Props
+
+<ArgsTable story="Overview" />
+
+## Usage
+
+Import and use the `DataGrid` component in your components as follows.
+
+<Source language="jsx" dark format code={dedent`import DataGrid from '@oxygen-ui/react/DataGrid';\n`} />
```
Comment (savindi7) [inline packages/react/src/components/DataGrid/DataGrid.stories.mdx:73]: Let's add a demo function too to show the usage with an example in the Story.

Refer: https://github.com/wso2/oxygen-ui/blob/main/packages/react/src/components/MenuItem/MenuItem.stories.mdx#L52

File: packages/react/src/components/DataGrid/DataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {DataGrid as MuiDataGrid, DataGridProps as MuiDataGridProps} from '@mui/x-data-grid';
+
```
Comment (savindi7) [inline packages/react/src/components/DataGrid/DataGrid.tsx:20]:
```suggestion
```

File: packages/react/src/components/DataGrid/DataGrid.stories.mdx (hunk 1 above)
Comment (ChanikaRuchini) [reply packages/react/src/components/DataGrid/DataGrid.stories.mdx]: Fixed

File: packages/react/src/components/DataGrid/DataGrid.tsx (hunk 3 above)
Comment (ChanikaRuchini) [reply packages/react/src/components/DataGrid/DataGrid.tsx]: Fixed

File: packages/react/src/components/DataGrid/DataGrid.stories.mdx (hunk 2 above)
Comment (ChanikaRuchini) [reply packages/react/src/components/DataGrid/DataGrid.stories.mdx]: Fixed

**Review by savindi7 — APPROVED** [review]

**Pull request merged**